internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/server/VeniceServerTest.java

```diff
@@ -209,6 +209,9 @@ public void testStartServerAndShutdownWithPartitionAssignmentVerification() {
       repository = server.getVeniceServer().getStorageService().getStorageEngineRepository();
       Assert.assertEquals(repository.getAllLocalStorageEngines().size(), 1);
       Assert.assertEquals(storageService.getStorageEngine(storeName).getPartitionIds().size(), 0);
+      server.getVeniceServer().getStorageService().removeStorageEngine(storeName);
```

Comment:
This logic should be done in `checkWhetherStoragePartitionsShouldBeKeptOrNot`. 

If we've deleted all partitions for a `storageEngine`, we should call `removeStorageEngine`.

Reply:
Since `removeStorageEngine`, is now called in `checkWhetherStoragePartitionsShouldBeKeptOrNot`, you can remove line 212. Once that's in I'll kick it off to Zac.